src/PSRule.Rules.Azure/rules/Azure.VNET.Rule.ps1

```diff
@@ -120,6 +120,28 @@ Rule 'Azure.VNET.FirewallSubnet' -Ref 'AZR-000322' -Type 'Microsoft.Network/virt
     $Assert.In($subnets, '.', @('AzureFirewallSubnet')).ReasonFrom('properties.subnets', $LocalizedData.SubnetNotFound, 'AzureFirewallSubnet')
 }
 
+# Synopsis: Disable default outbound access.
+Rule 'Azure.VNET.PrivateSubnet' -Ref 'AZR-000447' -Type 'Microsoft.Network/virtualNetworks', 'Microsoft.Network/virtualNetworks/subnets' -Tag @{ release = 'preview'; ruleSet = '2024_09'; 'Azure.WAF/pillar' = 'Security'; } {
+    if ($PSRule.TargetType -eq 'Microsoft.Network/virtualNetworks') {
+        $subnets = @(
+            $TargetObject.properties.subnets | Where-Object { -not $_.properties.delegations }
+            GetSubResources -ResourceType 'Microsoft.Network/virtualNetworks/subnets' | Where-Object { -not $_.properties.delegations }
+        )
```

Comment:
@BenjaminEngeset You're getting a null because the `properties.subnets` property might not exist. When null is passed though it the pipeline is also passes `-not $_.properties.delegations` which leaves `$subnets.Count` = 1.

```suggestion
        $subnets = @(
            $TargetObject.properties.subnets | Where-Object { $Null -ne $_ -and -not $_.properties.delegations }
            GetSubResources -ResourceType 'Microsoft.Network/virtualNetworks/subnets' | Where-Object { $Null -ne $_ -and -not $_.properties.delegations }
        )
```

Reply:
Thanks, @BernieWhite. I totally forgot that PowerShell returns `$null` as a placeholder to represent the not existing property.